src/components/Avatar/Avatar.tsx

```diff
@@ -2,14 +2,56 @@ import * as PropTypes from 'prop-types'
 import * as React from 'react'
 import { Image, Label, Icon } from '../../'
 
-import { customPropTypes, UIComponent } from '../../lib'
+import { customPropTypes, UIComponent, NotStrictProps } from '../../lib'
+import { ComponentVariablesInput, IComponentPartStylesInput } from '../../../types/theme'
+
+export interface IAvatarPropsStrict {
+  /** The alternative text for the image used in the Avatar. */
+  alt?: string
+
+  /** An element type to render as (string or function). */
+  as?: any
+
+  /** Additional classes. */
+  className?: string
+
+  /** The name used for displaying the initials of the avatar if the image is not provided. */
+  name?: string
```

Comment:
should we consider to extract these props into `IBaseProps`:
```TS
/** An element type to render as (string or function). */
  as?: any
   /** Additional classes. */
  className?: string
   /** The name used for displaying the initials of the avatar if the image is not provided. */
  name?: string
```
so, it looks like:
```TS
interface IBaseProps {
/** An element type to render as (string or function). */
  as?: any
   /** Additional classes. */
  className?: string
   /** The name used for displaying the initials of the avatar if the image is not provided. */
  name?: string
}

export interface IAvatarPropsStrict extends IBaseProps {
...
```